File: examples/python/03_search_listings.py
"""Bounding-box search for active for-sale listings.

`/v1/listings` (GET) accepts a comma-separated `bbox` string in decimal degrees
("west,south,east,north"). The bbox below covers central San Francisco.

Search results use a different field shape than property detail records:
- `address` is a string (not an object)
- `beds` / `baths` (not `bedrooms` / `bathrooms`)
- `unformattedPrice` is the integer; `price` is the pre-formatted display string
- `area` is the living-area sqft

Run:
    export ZILLAPI_KEY="zk_..."
    python 03_search_listings.py
"""
import os

import httpx

KEY = os.environ["ZILLAPI_KEY"]
BBOX = "-122.45,37.74,-122.40,37.79"

resp = httpx.get(
    "https://api.zillapi.com/v1/listings",
    params={
        "status": "for_sale",
        "bbox": BBOX,
        "price_min": 800_000,
        "price_max": 2_000_000,
        "beds_min": 2,
        "home_types": "house,condo,townhouse",
        "max_items": 20,
    },
    headers={"Authorization": f"Bearer {KEY}"},
    timeout=60,
)
resp.raise_for_status()
listings = resp.json()["data"]

print(f"Found {len(listings)} listings\n")
for row in listings:
    print(
        f"  {row.get('address', '?'):40s}  "
        f"${row.get('unformattedPrice') or 0:>10,}  "
        f"{row.get('beds', '?')}bd/{row.get('baths', '?')}ba  "
        f"{row.get('area') or 0:,} sqft"
    )

File: examples/node/03_search_listings.mjs
// Bounding-box search for active for-sale listings.
//
// `/v1/listings` (GET) accepts a comma-separated `bbox` string in decimal
// degrees ("west,south,east,north"). The bbox below covers central San
// Francisco.
//
// Search results use a different field shape than property detail records:
// `address` is a string, `beds`/`baths` (not `bedrooms`/`bathrooms`),
// `unformattedPrice` is the integer.
//
// Run:
//   export ZILLAPI_KEY="zk_..."
//   node 03_search_listings.mjs

const KEY = process.env.ZILLAPI_KEY;
if (!KEY) {
  console.error("Set ZILLAPI_KEY env var: export ZILLAPI_KEY=zk_...");
  process.exit(1);
}

const url = new URL("https://api.zillapi.com/v1/listings");
url.searchParams.set("status", "for_sale");
url.searchParams.set("bbox", "-122.45,37.74,-122.40,37.79");
url.searchParams.set("price_min", "800000");
url.searchParams.set("price_max", "2000000");
url.searchParams.set("beds_min", "2");
url.searchParams.set("home_types", "house,condo,townhouse");
url.searchParams.set("max_items", "20");

const resp = await fetch(url, {
  headers: { Authorization: `Bearer ${KEY}` },
});

if (!resp.ok) {
  throw new Error(`HTTP ${resp.status}: ${await resp.text()}`);
}

const { data } = await resp.json();
console.log(`Found ${data.length} listings\n`);

for (const row of data) {
  const price = row.unformattedPrice
    ? `$${row.unformattedPrice.toLocaleString()}`
    : row.price ?? "n/a";
  console.log(
    `  ${(row.address ?? "?").padEnd(40)}  ${price.padStart(12)}  ` +
      `${row.beds ?? "?"}bd/${row.baths ?? "?"}ba  ${row.area?.toLocaleString() ?? "?"} sqft`
  );
}
